# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerScope.h
# Commit: 5598943a94fd
# Full Hash: 5598943a94fd9e6be45bbc7125ef7bac49a4bf05
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2021-12-15 21:51:13
# Regressor Bug: 1744025
# File Overlap Count: 3
# Description:
#   Bug 1744025: Use CheckedUnsafePtr<WorkerPrivate> in WorkerGlobalScopeBase. r=dom-worker-reviewers,asuth
#   
#   Depends on D132708
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D132709
# ==============================================================================

diff -r 43186fbbf8b4 -r 5598943a94fd dom/workers/WorkerScope.h
--- a/dom/workers/WorkerScope.h	Wed Dec 15 13:48:17 2021 +0000
+++ b/dom/workers/WorkerScope.h	Wed Dec 15 13:48:17 2021 +0000
@@ -20,6 +20,7 @@
 #include "mozilla/dom/ImageBitmapBinding.h"
 #include "mozilla/dom/ImageBitmapSource.h"
 #include "mozilla/dom/SafeRefPtr.h"
+#include "mozilla/dom/WorkerPrivate.h"
 #include "nsCOMPtr.h"
 #include "nsCycleCollectionParticipant.h"
 #include "nsIGlobalObject.h"
@@ -93,7 +94,7 @@
   NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_CLASS_INHERITED(WorkerGlobalScopeBase,
                                                          DOMEventTargetHelper)
 
-  WorkerGlobalScopeBase(NotNull<WorkerPrivate*> aWorkerPrivate,
+  WorkerGlobalScopeBase(WorkerPrivate* aWorkerPrivate,
                         UniquePtr<ClientSource> aClientSource);
 
   virtual bool WrapGlobalObject(JSContext* aCx,
@@ -162,7 +163,7 @@
  protected:
   ~WorkerGlobalScopeBase();
 
-  const NotNull<WorkerPrivate*> mWorkerPrivate;
+  CheckedUnsafePtr<WorkerPrivate> mWorkerPrivate;
 
  private:
   RefPtr<Console> mConsole;
@@ -344,7 +345,7 @@
   NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_CLASS_INHERITED(
       DedicatedWorkerGlobalScope, WorkerGlobalScope)
 
-  DedicatedWorkerGlobalScope(NotNull<WorkerPrivate*> aWorkerPrivate,
+  DedicatedWorkerGlobalScope(WorkerPrivate* aWorkerPrivate,
                              UniquePtr<ClientSource> aClientSource,
                              const nsString& aName);
 
@@ -388,7 +389,7 @@
     : public WorkerGlobalScope,
       public workerinternals::NamedWorkerGlobalScopeMixin {
  public:
-  SharedWorkerGlobalScope(NotNull<WorkerPrivate*> aWorkerPrivate,
+  SharedWorkerGlobalScope(WorkerPrivate* aWorkerPrivate,
                           UniquePtr<ClientSource> aClientSource,
                           const nsString& aName);
 
@@ -410,8 +411,7 @@
                                            WorkerGlobalScope)
 
   ServiceWorkerGlobalScope(
-      NotNull<WorkerPrivate*> aWorkerPrivate,
-      UniquePtr<ClientSource> aClientSource,
+      WorkerPrivate* aWorkerPrivate, UniquePtr<ClientSource> aClientSource,
       const ServiceWorkerRegistrationDescriptor& aRegistrationDescriptor);
 
   bool WrapGlobalObject(JSContext* aCx,